src/agreement/config.ts

```diff
@@ -35,12 +36,14 @@ export class AgreementConfig {
 export class AgreementServiceConfig extends AgreementConfig {
   readonly agreementSelector: AgreementSelector;
   readonly agreementMaxEvents: number;
+  readonly agreementMaxPoolSize: number;
   readonly agreementEventsFetchingIntervalSec: number;
 
   constructor(options?: AgreementServiceOptions) {
     super(options);
     this.agreementSelector = options?.agreementSelector ?? DEFAULTS.agreementSelector;
     this.agreementMaxEvents = options?.agreementMaxEvents ?? DEFAULTS.agreementMaxEvents;
+    this.agreementMaxPoolSize = options?.agreementMaxPoolSize ?? DEFAULTS.agreementMaxPoolSize;
```

Comment:
What is the benefit of setting `agreementMaxPoolSize` to anything other than the same value as `maxParallelTasks`? i think these two config values should just be merged to one

Reply:
In my opinion, AgreementPool does not know what a task is and does not know the concept of maxParallelTasks. In the context of agreemnt domain, it is limited only to the pool size parameter. The TaskExecutor maps maxParallelTask to the max size of the agreement pool. And since you're asking, a comment explaining why would probably be useful here. so I'll add a comment here.

We will soon get rid of TaskExecutor from this repo and whatever uses AgreementPool should set a parameter related to its context.

Maybe agreementMaxPoolSize is not the best name, but e.g. if it will be used by a service or deployment (in our new API), maxParallelTask will not fit..

Reply:
You're right, when reading the PR I missed that the default is `DEFAULTS.maxParallelTasks`. You can disregard this comment then